src/pybind/mgr/cephadm/tests/test_cephadm.py

```diff
@@ -476,6 +478,29 @@ def test_extra_container_args(self, _run_cephadm, cephadm_module: CephadmOrchest
                     image='',
                 )
 
+    @mock.patch("cephadm.serve.CephadmServe._run_cephadm")
+    def test_extra_entrypoint_args(self, _run_cephadm, cephadm_module: CephadmOrchestrator):
+        _run_cephadm.side_effect = async_side_effect(('{}', '', 0))
+        with with_host(cephadm_module, 'test'):
+            with with_service(cephadm_module, ServiceSpec(service_type='node-exporter',
+                              extra_entrypoint_args=['--collector.textfile.directory=/var/lib/node_exporter/textfile_collector', '--some-other-arg']),
+                              CephadmOrchestrator.apply_node_exporter):
+                _run_cephadm.assert_called_with(
+                    'test', 'node-exporter.test', 'deploy', [
+                        '--name', 'node-exporter.test',
+                        '--meta-json', ('{"service_name": "node-exporter", "ports": [9100], "ip": null, "deployed_by": [], "rank": null, '
+                                        '"rank_generation": null, "extra_container_args": null, "extra_entrypoint_args": '
+                                        '["--collector.textfile.directory=/var/lib/node_exporter/textfile_collector", '
+                                        '"--some-other-arg"]}'),
+                        '--config-json', '-',
+                        '--tcp-ports', '9100',
+                        '--extra-entrypoint-args=--collector.textfile.directory=/var/lib/node_exporter/textfile_collector',
+                        '--extra-entrypoint-args=--some-other-arg'
```

Comment:
Since we are adding UT it would be good to have both `extra_container_args` and `extra_entrypoint_args` and make sure both of them are used in the call correctly.

Reply:
a unit test that intermixes both container and entrypoint args would be nice!

Reply:
a unit test that intermixes the use of container and entrypoint args would be nice!

Reply:
Now that I see how these values are being passed to cephadm, I'm concerned about whitespace/argument splitting. Can we also add test cases that include multiple arguments and some that have spaces in the values?

Reply:
added a unit that does both together. @phlogistonjohn using spaces didn't work and hasn't worked with extra_container_args either since its inception (see https://tracker.ceph.com/issues/57338 and the commit message of the new commit on this PR).  I decided I really ought to just fix the issue here, so now there's a unit test with spaces for extra_container_args and extra_entrypoint_args together and it should actually work.

Reply:
Works for me, thanks!  That new patch seems reasonable to me. I'll try to (re)review the whole PR by tomorrow!